### DISCUSSION TITLE
[gatsby-source-shopify] Get product ID

### DISCUSSION THREAD

**Discussion by code-mat** [gatsbyjs/gatsby#29517] [Help] [answered]
## Description

Hi there. I'm using `gatsby-source-shopify` to extract API data from Shopify. Currently it's not possible to get the actual product ID from the query. We have only two fields which are prefixed:

- id: Gatsby's internal ID
- shopifyId: storefrontId

![Screenshot 2021-02-12 at 18 05 24](https://user-images.githubusercontent.com/13292060/107798942-15334600-6d5d-11eb-8502-34b2d1b3ed14.png)

### Expected result

I'd need to pull a direct product ID like this: `6179957932200`

**Comment by LekoArts** [comment, marked as answer]
Please see https://community.shopify.com/c/Shopify-APIs-SDKs/Storefront-API-variantId-different-than-Rest-API-variantId/m-p/607819/highlight/true#M41291

You'll need to convert the ID:

```js
const storefrontId = "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0LzEwMDc5Nzg1MTAw";
const adminGraphqlId = atob(storefrontId);
console.log(adminGraphqlId) // prints "gid://shopify/Product/10079785100"
```

**Reply by code-mat** [reply]
thanks man!!! it's super handy 👍
